asset_dashboard/static/js/components/map_utils/MapClipper.js

```diff
@@ -52,22 +55,70 @@ export default function MapClipper({ geoJson, onClipped }) {
     })
   }
 
-  function clipGeometries(geometries) {
-    if (geometries) {
-      const bounds = {
-        'type': 'FeatureCollection',
-        'features': Object.values(geometries)
-      }
-  
-      const clippedFeatureCollection = clip(bounds, geoJson)
-  
-      if (clippedFeatureCollection.features.length > 0) {
-        onClipped(clippedFeatureCollection)
+  /* Separate the geometry types. We need the LineStrings to clip, 
+    and separately all of the other geometry types to see if they 
+    intersect with the drawnGeometries. 
+    A LineString would be a trail that can be clipped.
+    The remaining types would be like a building or structure, etc
+    that should be treated as an entire entity and not clipped. */
+  function separateGeometryTypes() {
+    const lineStringFeatureTypes = []
+    const allOtherFeatureTypes = []
+    
+    turf.featureEach(geoJson, (currentFeature) => {
```

Comment:
TIL, cool method.

Reply:
TIAL [for loops are faster](https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/) than declarative methods like `Array.filter`.

Reply:
interesting! i didn't know that either. i like the syntax of `array.map(item => ...)` because it's easier for me to grok when reading it. especially compared to python's similar list methods like `map` etc. but that's good info to have whenever performance is an issue.